custom/icds_reports/tasks.py

```diff
@@ -1283,3 +1283,17 @@ def _dictfetchall(cursor):
         dict(zip(columns, row))
         for row in cursor.fetchall()
     ]
+
+def setup_aggregation(agg_date):
+    _update_ucr_table_mapping()
+
+    db_alias = get_icds_ucr_db_alias_or_citus(False)
+    if db_alias:
+        with connections[db_alias].cursor() as cursor:
+            _create_aggregate_functions(cursor)
+
+        _update_aggregate_locations_tables()
+    state_ids = list(SQLLocation.objects
+                     .filter(domain=DASHBOARD_DOMAIN, location_type__name='state')
+                     .values_list('location_id', flat=True))
+    cache.set('agg_state_ids_{}'.format(agg_date), state_ids, ONE_DAY * 2)
```

Comment:
This branch is based off of a commit 4 weeks old. This will break in prod because `cache` was removed almost 3 weeks ago https://github.com/dimagi/commcare-hq/pull/24777/commits/2fd77150c4c03f1ad9f425b255d4ef8c182b90f5

Reply:
What's the benefit of putting this key in redis? We could remove the dependency on redis by introducing a new table `IcdsStateAggregation` or something with columns `agg_uuid`, `state_id`, `date`. This would remove the dependency on redis, and provide a history of when states were begun being aggregated.